Source/Libraries/openXDA.Model/System/DataOperationFailure.cs

```diff
@@ -0,0 +1,50 @@
+﻿//******************************************************************************************************
+//  DataOperationFailure.cs - Gbtc
+//
+//  Copyright © 2017, Grid Protection Alliance.  All Rights Reserved.
+//
+//  Licensed to the Grid Protection Alliance (GPA) under one or more contributor license agreements. See
+//  the NOTICE file distributed with this work for additional information regarding copyright ownership.
+//  The GPA licenses this file to you under the MIT License (MIT), the "License"; you may
+//  not use this file except in compliance with the License. You may obtain a copy of the License at:
+//
+//      http://opensource.org/licenses/MIT
+//
+//  Unless agreed to in writing, the subject software distributed under the License is distributed on an
+//  "AS-IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. Refer to the
+//  License for the specific language governing permissions and limitations.
+//
+//  Code Modification History:
+//  ----------------------------------------------------------------------------------------------------
+//  07/24/2017 - Billy Ernest
+//       Generated original version of source code.
+//
+//******************************************************************************************************
+
+using System;
+using GSF.Data.Model;
+
+namespace openXDA.Model
+{
+    [PostRoles("Administrator, Transmission SME")]
+    [DeleteRoles("Administrator, Transmission SME")]
+    [PatchRoles("Administrator, Transmission SME")]
+    [AllowSearch]
```

Comment:
```suggestion
    [PostRoles("")]
    [PatchRoles("")]
    [DeleteRoles("Administrator, Transmission SME")]
    [AllowSearch]
```
Nitpicky, but I don't think these should be editable via a controller. Only written by the system

Reply:
Good catch!